SparkPost/2web2ui
# TR-1512 Duplicate Template

[TR-1512](https://jira.int.messagesystems.com/browse/TR-1512)

### What Changed
- Updated `<DraftModeActions/>` component, adding new `<DuplicateTemplate/>`
- Added new `<DuplicateTemplateModal/>` component that is outside of the existing `<Popover/>` to help address toggling behavior, keeping it independent from the Popover itself
- Extracted new `<SaveAndPublishConfirmationModal/>` component to address similar behavior with the `<SaveAndPublish/>` component
- Updated `<SaveAndPublish/>` component to avoid having nested `<button>` elements within `<a>` elements
- Updated some styling for the `<EditNavigation/>` component to mirror design specs more closely
- Updated relevant tests, wrote new tests
- Updated Matchbox to at least version `3.7.0` to allow use of the `<ScreenReaderOnly>` component
- Added `<ButtonWrapper>` generic component for consistent positioning of button elements within the UI - includes template, styles, and documentation
- Added screen reader only content to the `<FullPage/>` component to ensure screen reader users properly interpret the back button

### How To Test
1. Navigate to http://localhost:3100/templatesV2
1. Select a template that is in the draft status
1. Click on the popover icon to the right of the 'Save and Publish' button:

<img width="307" alt="Screen Shot 2019-09-05 at 10 42 01 AM" src="https://user-images.githubusercontent.com/3613392/64352326-d9257e80-cfc9-11e9-88f1-723e92f281da.png">

1. Click on 'Duplicate Template'
1. A modal should appear with extra copy appended to the relevant text fields (`(COPY)` for the name field or `-copy` for the ID field)
1. After confirming, the user should be re-routed to `/templatesV2`
1. The duplicated template should now appear on the list page

### To Do
- [ ] __QUESTION__ - if I duplicate a template that has been published previously, is the new duplicated template supposed to render in the table with a status of 'Published'?
